File: ros_ws/src/my_robot_tutorials/scripts/add_two_ints_client.py
#!/usr/bin/env python

import rospy
from rospy_tutorials.srv import AddTwoInts


def main(args=None):
    rospy.init_node("add_two_ints_client")
    rospy.loginfo("add_two_ints_client has been created")

    rospy.wait_for_service("add_two_ints")

    try:
        add_two_ints = rospy.ServiceProxy("add_two_ints", AddTwoInts)
        response = add_two_ints(9, 7)
        rospy.loginfo("Result is: " + str(response.sum))
    except rospy.ServiceException as e:
        rospy.logwarn("Service warning: " + str(e))


if __name__ == "__main__":
    main()

File: ros_ws/src/my_robot_tutorials/src/add_two_ints_client.cpp
#include <ros/ros.h>
#include <rospy_tutorials/AddTwoInts.h>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "add_two_ints_client");
    ros::NodeHandle nh;
    ros::ServiceClient client = nh.serviceClient<rospy_tutorials::AddTwoInts>("add_two_ints");
    
    ROS_INFO("add_two_ints_client has been created");

    rospy_tutorials::AddTwoInts srv;
    srv.request.a = 9;
    srv.request.b = 7;

    if(client.call(srv)) {
        ROS_INFO("%d + %d = %d", (int)srv.request.a, (int)srv.request.b, (int)srv.response.sum);
    }
    else {
        ROS_WARN("Service call failed");
    }

    return 0;
}